engine/src/cmd/movable.cpp

```diff
@@ -466,3 +470,362 @@ double Movable::GetMaxWarpFieldStrength(float rampmult) const {
+    if ((unit->afterburntype == 2) && unit->warpenergy < 0) {
+        unit->warpenergy = 0;
+        afterburn = false;
+    }
+    if (3 == unit->afterburntype) {      //no afterburner -- we should really make these types an enum :-/
+        afterburn = false;
+    }
+    Vector Res = amt1;
+
+    float fuelclamp = (unit->fuel <= 0) ? configuration()->fuel.no_fuel_thrust : 1;
+    float abfuelclamp = (unit->fuel <= 0) ? configuration()->fuel.no_fuel_afterburn : 1;
+    if (fabs(amt1.i) > fabs(fuelclamp * limits.lateral)) {
+        Res.i = copysign(fuelclamp * limits.lateral, amt1.i);
+    }
+    if (fabs(amt1.j) > fabs(fuelclamp * limits.vertical)) {
+        Res.j = copysign(fuelclamp * limits.vertical, amt1.j);
+    }
+    float ablimit =
+            afterburn
+                    ? ((limits.afterburn - limits.forward) * abfuelclamp + limits.forward * fuelclamp)
+                    : limits.forward;
+    if (amt1.k > ablimit) {
+        Res.k = ablimit;
+    }
+    if (amt1.k < -limits.retro) {
+        Res.k = -limits.retro;
+    }
+    const float Lithium6constant = configuration()->fuel.deuterium_relative_efficiency_lithium;
+    //1/5,000,000 m/s
+    const float FMEC_exit_vel_inverse = configuration()->fuel.fmec_exit_velocity_inverse;
+    if (unit->afterburntype == 2) {
+        //Energy-consuming afterburner
+        //HACK this forces the reaction to be Li-6+Li-6 fusion with efficiency governed by the getFuelUsage function
+        unit->warpenergy -= unit->afterburnenergy * Energetic::getFuelUsage(afterburn) * simulation_atom_var * Res.Magnitude()
+                * FMEC_exit_vel_inverse
+                / Lithium6constant;
+    }
+    if (3 == unit->afterburntype || unit->afterburntype == 1) {
+        //fuel-burning overdrive - uses afterburner efficiency. In NO_AFTERBURNER case, "afterburn" will always be false, so can reuse code.
+        //HACK this forces the reaction to be Li-6+Li-6 fusion with efficiency governed by the getFuelUsage function
+        unit->fuel -=
+                ((afterburn
+                        && finegrainedFuelEfficiency) ? unit->afterburnenergy : Energetic::getFuelUsage(afterburn))
+                        * simulation_atom_var * Res.Magnitude()
+                        * FMEC_exit_vel_inverse / Lithium6constant;
+#ifndef __APPLE__
+        if (ISNAN(unit->fuel)) {
+            VS_LOG(error, "Fuel is NAN A");
+            unit->fuel = 0;
+        }
+#endif
+    }
+    if (unit->afterburntype == 0) {
+        //fuel-burning afterburner - uses default efficiency - appears to check for available energy? FIXME
+        //HACK this forces the reaction to be Li-6+Li-6 fusion with efficiency governed by the getFuelUsage function
+        unit->fuel -= unit->getFuelUsage(false) * simulation_atom_var * Res.Magnitude() * FMEC_exit_vel_inverse / Lithium6constant;
+#ifndef __APPLE__
+        if (ISNAN(unit->fuel)) {
+            VS_LOG(error, "Fuel is NAN B");
+            unit->fuel = 0;
+        }
+#endif
```

Comment:
Perhaps we need to change compiler flags instead of opting out platforms from this logic check?
https://developer.apple.com/forums/thread/656056

Reply:
I dug a bit into this and I don't know why it's here. The original [commit](https://github.com/vegastrike/Vega-Strike-Engine-Source/commit/58be34265c8382c848eadfce58aea7d35cd391e0) is from 2006. OSX was introduced in 2001. Apple switched to Intel computers in 2006 but that should have made it more consistent with Linux, not break it. There's a good chance that if we get the game running on Mac, we won't need this.

Reply:
@royfalk if we can get rid of it that'd be awesome!

Reply:
@BenjamenMeyer I'd like to recommend that we leave this code as is for now. Then once we get #773 reviewed and merged (hint, hint) then we can come back and revisit this topic.

Reply:
@stephengtuggy that's fine. It'd be nice to simplify this so it's the same on all platforms or easily swappable via compiler flags instead.  However we get there - or whichever PR gets us there - doesn't matter to me. Thanks!

Reply:
Moved to #787. Resolving conversation.